todo_app/app.py

```diff
@@ -1,40 +1,282 @@
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        scheme = environ.get('HTTP_X_FORWARDED_PROTO')
+        if scheme:
+            environ['wsgi.url_scheme'] = scheme
+        return self.app(environ, start_response)
+
+
+class UserToLogin(UserMixin):
+    def __init__(self, id):
+        self.id = id
 
 
 def create_app():
     app = Flask(__name__)
-    app.config.from_object(Config)
-    trello.init()
+    app.config.from_object('todo_app.flask_config.Config')
+    app.wsgi_app = ReverseProxied(app.wsgi_app)
+    app.logger.setLevel(app.config['LOG_LEVEL'])
+    if app.config['LOGGLY_TOKEN'] is not None:
+        handler = HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app')
+        handler.setFormatter(Formatter("[%(asctime)s] %(levelname)s in %(module)s: %(message)s"))
+        app.logger.addHandler(handler)
+
+    client_id = os.environ['GITHUB_CLIENT_ID']
+    client_secret = os.environ['GITHUB_CLIENT_SECRET']
+    base_url = "https://api.github.com"
+    authorization_url = "https://github.com/login/oauth/authorize"
+    token_endpoint = "https://github.com/login/oauth/access_token"
+
+    client = WebApplicationClient(client_id)
+    todo = DbConnectionManager()
+    user_manager = DbUserManager()
+
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        app.logger.debug("Loading user for login......")
+        return UserToLogin(user_id)
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        app.logger.debug("User is not authenticated, redirect to login......")
+        return redirect(url_for('login'))
+
+    @app.route('/logout')
+    @login_required
+    def logout():
+        logout_user()
+        app.logger.info("User logged out successfully")
+        session.clear()
+        return redirect("https://github.com/logout")
+
+    @app.route("/login")
+    def login():
+        request_uri = client.prepare_request_uri(
+            authorization_url,
+            redirect_uri=request.base_url + "/callback",
+            scope=None,
+        )
+        return redirect(request_uri)
+
+    @app.route("/login/callback")
+    def callback():
+        code = request.args.get("code")
+
+        # Prepare and send request to get tokens!
+        token_url, headers, body = client.prepare_token_request(
+            token_endpoint,
+            authorization_response=request.url,
+            redirect_url=request.base_url,
+            code=code,
+        )
+
+        token_response = requests.post(
+            token_url,
+            headers=headers,
+            data=body,
+            auth=(client_id, client_secret),
+        )
+        app.logger.debug("Token url:", token_url)
+        if token_response.status_code != 200:
+            app.logger.critical("Failed to get Auth Token, redirecting to login screen")
```

Comment:
I don't think this needs to be `critical`, `error` seems more appropriate. Critical is usually reserved for errors which significantly impact the system rather than a problem for a specific user/request.